workers/unity/Packages/com.improbable.gdk.querybasedinteresthelper/Constraint.cs

```diff
@@ -0,0 +1,388 @@
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+namespace Improbable.Gdk.QueryBasedInterest
+{
+    /// <summary>
+    ///     Utility class to help construct QueryConstraint objects.
+    /// </summary>
+    public static class Constraint
```

Comment:
I see we have a lot of methods that take a generated QueryConstraint as a parameter. Would it make more sense to have this as a non static class use that instead. Otherwise a user may try to use our helpers, not realise this is here and then end up constructing them from the base objects anyway.